services/payments/src/simcore_service_payments/api/rest/_acknowledgements.py

```diff
@@ -19,34 +29,82 @@
+
+    with log_context(
+        _logger,
+        logging.INFO,
+        "Top-up %s credits for %s in RUT",
+        f"{transaction.osparc_credits}",
+        f"{transaction.payment_id=}",
+    ):
+        credit_transaction_id = await rut_api.create_credit_transaction(
+            product_name=transaction.product_name,
+            wallet_id=transaction.wallet_id,
+            wallet_name="id={transaction.wallet_id}",
+            user_id=transaction.user_id,
+            user_email=transaction.user_email,
+            osparc_credits=transaction.osparc_credits,
+            payment_transaction_id=transaction.payment_id,
+            created_at=transaction.completed_at,
+        )
+
     _logger.debug(
-        "payment completed: %s",
-        f"{payment_id=}, {ack.success=}, {ack.message=}, {session.username=}",
+        "RUT response to %s was %s",
+        f"{transaction.payment_id=}",
+        f"{credit_transaction_id=}",
     )
-    _logger.debug("Notify front-end -> sio ")
-    _logger.debug("Authorize inc/dec credits -> RUT")
-    _logger.debug("Annotate RUT response")
 
 
 @router.post("/payments/{payment_id}:ack")
 async def acknowledge_payment(
     payment_id: PaymentID,
     ack: AckPayment,
-    session: Annotated[SessionData, Depends(get_current_session)],
+    _session: Annotated[SessionData, Depends(get_current_session)],
+    repo: Annotated[
+        PaymentsTransactionsRepo, Depends(get_repository(PaymentsTransactionsRepo))
+    ],
+    rut_api: Annotated[ResourceUsageTrackerApi, Depends(get_rut_api)],
     background_tasks: BackgroundTasks,
 ):
     """completes (ie. ack) request initated by `/init` on the payments-gateway API"""
-    _logger.debug(
-        "User %s is acknoledging payment with %s as %s", session, f"{payment_id=}", ack
-    )
-    _logger.debug("Validate and complete transaction -> DB")
-    _logger.debug(
-        "When annotated in db, respond and start a background task with the rest"
-    )
-    background_tasks.add_task(on_payment_completed, payment_id, ack, session)
-    raise HTTPException(status_code=status.HTTP_501_NOT_IMPLEMENTED)
+
+    with log_context(
+        _logger,
+        logging.INFO,
+        "Annotate ACK transaction %s in db",
+        f"{payment_id=}",
+    ):
+        try:
+            transaction = await repo.update_ack_payment_transaction(
+                payment_id=payment_id,
+                completion_state=(
+                    PaymentTransactionState.SUCCESS
+                    if ack.success
+                    else PaymentTransactionState.FAILED
+                ),
+                state_message=ack.message,
+                invoice_url=ack.invoice_url,
+            )
+        except PaymentNotFoundError as err:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND, detail=f"{err}"
+            ) from err
+
+    if ack.saved:
+        _logger.debug("TODO Annotate CREATE payment method")
```

Comment:
TODO inside a log? that is quite the trick...

Reply:
and my comment from above applies here as well

Reply:
well I think that if you put a TODO in the code and fix it in the next PR that is quite ok, no need to go in these tricks. if you only think you will fix it in the "near future"... I think this is a good candidate to never being implemented. Now, say you put the comment in a TODO, at least the next developer that edit the file will be annoyed by sonarcloud and might remove it... that is my 5 cents..